manual/manual/html_processing/js/search.js

```diff
@@ -0,0 +1,242 @@
+	indexState = 'LOADING';
+	
+	const script = document.createElement('script');
+	script.src = 'index.js';
+	script.addEventListener('load', () => {
+	    indexState = 'HAS_LOADED';
+	    mySearch(includeDescr);
+	});
+	document.head.appendChild(script);
+	return true;
+	
+    case 'LOADING':
+	return true;
+	
+    case 'HAS_LOADED':
+	return false;
+    }
+}
+
+// check if sub is a substring of s. The start/end of the string s are
+// marked by "^" and "$", and hence these chars can be used in sub to
+// refine the search.
+function isSubString (sub, s) {
+    // s = s.toLowerCase();
+    // sub = sub.toLowerCase();
+    // case sensitive is better for OCaml modules.
+    let ss = "^" + s + "$";
+    return (ss.indexOf(sub) !== -1);
+}
+
+// line is a string array. We check if sub is a substring of one of
+// the elements of the array.
+function hasSubString (sub, line) {
+    return (line.findIndex(function (s) {
+	return (isSubString(sub,s)); }) !== -1);
+}
+
+// Check if one of the strings in subs is a substring of one of the
+// strings in line.
+function hasSubStrings (subs, line) {
+    return (subs.findIndex(function (sub) {
+	return (hasSubString (sub, line)); }) !== -1);
+}
+
+// error of sub being a substring of s. Best if starts at 0. Except
+// for strings containing "->", which is then best if the substring is
+// at the most right-hand position (representing the "return type").
+// markers "^" and "$" for start/end of string can be used: if they
+// are not satisfied, the MAX_ERROR is returned.
+function subError (sub, s) {
+    let StartOnly = false;
+    let EndOnly = false;
+    if (sub.length>1) {
+	if (sub[0] == "^") {
+	    StartOnly = true;
+	    sub = sub.substring(1);
+	}
+	if (sub[sub.length - 1] == "$") {
+	    EndOnly = true;
+	    sub = sub.substring(0, sub.length - 1);
+	}
+    }
+    let err = s.indexOf(sub);
+    if (err == -1 ||
+	(StartOnly && err != 0) ||
+	(EndOnly && err != s.length - sub.length)) {
+	err = MAX_ERROR;
+    } else {
+	if ( sub.includes("->") ) {
+	    err = Math.min(s.length - sub.length - err,1); // 0 or 1
+	    // err = 0 if the substring is right-aligned
+	} else {
+	    err = Math.min(err,1); // 0 or 1
+	    // err = 0 if the substring
+	}
+	err += Math.abs((s.length - sub.length) / s.length);}
+    return (err)
+    // between 0 and 2, except if MAX_ERROR
+}
+
+// Minimal substring error. In particular, it returns 0 if the string
+// 'sub' has an exact match with one of the strings in 'line'.
+function subMinError (sub, line) {
+    //let strings = [line[0], line[1], line[2]];
+    let errs = line.map(function (s) { return subError (sub, s); });
+    return Math.min(...errs); // destructuring assignment
+}
+
+
+function add (acc, a) {
+    return acc + a;
+}
+
+// for each sub we compute the minimal error within 'line', and then
+// take the average over all 'subs'. Thus it returns 0 if each sub has
+// an exact match with one of the strings in 'line'.
+function subsMinError (subs, line) {
```

Comment:
If this is an average, the name should be `subsAverageError` not `subsMinError`.

Reply:
let's do subsAvgMinError then